File: fftl-02-back/src/board/dto/saveBoard.dto.ts
import { Comment } from 'src/entities/comments.entity';
import { User } from 'src/entities/user.entity';

export class SaveBoardDto {
  readonly bid: number;
  readonly title: string;
  readonly bregdate: string;
  readonly nickname: string;
  readonly content: string;
  readonly user: User;
  readonly comments: Comment[];
}

File: fftl-02-back-Spring/src/main/java/fftl/fftl02backSpring/request/SaveBoardDto.java
package fftl.fftl02backSpring.request;

import fftl.fftl02backSpring.entity.Board;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SaveBoardDto {

    private String title;
    private String content;
    private String nickname;
    private String regdate;
    private Long user_id;


    public Board toEntity(){
        return Board.builder()
                .title(title)
                .content(content)
                .nickname(nickname)
                .bregdate(regdate)
                .build();
    }
}
